aesara/compile/profiling.py

```diff
@@ -37,93 +49,92 @@
 
 
 def _atexit_print_fn():
-    """
-    Print ProfileStat objects in _atexit_print_list to _atexit_print_file.
-
-    """
+    """Print `ProfileStat` objects in `_atexit_print_list` to `_atexit_print_file`."""
     if config.profile:
         to_sum = []
 
         if config.profiling__destination == "stderr":
-            destination_file = sys.stderr
+            destination_file = "<stderr>"
         elif config.profiling__destination == "stdout":
-            destination_file = sys.stdout
+            destination_file = "<stdout>"
         else:
-            destination_file = open(config.profiling__destination, "w")
-
-        # Reverse sort in the order of compile+exec time
-        for ps in sorted(
-            _atexit_print_list, key=lambda a: a.compile_time + a.fct_call_time
-        )[::-1]:
-            if (
-                ps.fct_callcount >= 1
-                or ps.compile_time > 1
-                or getattr(ps, "callcount", 0) > 1
-            ):
-                ps.summary(
+            destination_file = config.profiling__destination
+
+        with extended_open(destination_file, mode="w"):
```

Comment:
@brandonwillard I think you need to use the yielded value here.  Eg. `with extended_open(...) as f`.  As it is the file object is dropped and the string is passed through.